nemo/collections/nlp/models/language_modeling/megatron_gpt_sft_model.py

```diff
@@ -359,179 +384,206 @@ def fwd_bwd_step(self, dataloader_iter, batch_idx, forward_only):
-            "use_greedy": True,
-            "temperature": 1.0,
-            "top_k": 1,
-            "top_p": 0.94,
-            "repetition_penalty": 1.2,
-            "add_BOS": False,
-            "all_probs": False,
-            "compute_logprob": False,
-            "end_strings": ["<|endoftext|>"],
+            'preds': preds_text,  # [str]
+            'labels': labels_text,  # [str]
+            'inputs': inputs_text,  # [str]
+            'metadata': metadata,  # [dict]
         }
-        result = megatron_gpt_generate(
-            model=self,
-            inputs=(
-                batch['tokens'].cuda(),
-                (batch['context_lengths'] - 1).cuda(),
-            ),  # NOTE: We do -1 here to remove the space between context and response.
-            tokenizer=self.tokenizer,
-            sampling_params=sampling_params,
-            length_params=length_params,
-            check_sequence_parallel_and_checkpointing=False,  # We need to skip these checks since we'll manually enbale and disable checkpointing between training and validation.
-        )
 
-        preds_text = []
-        labels_text = []
-        input_text = []
-        for idx, item in enumerate(result['token_ids']):
-            pred = self.tokenizer.ids_to_text(item[batch['context_lengths'][idx] - 1 :])
-            input = self.tokenizer.ids_to_text(item[: batch['context_lengths'][idx] - 1])
-            label = self.tokenizer.ids_to_text(batch['tokens'][idx][batch['context_lengths'][idx] :].tolist())
-            preds_text.append(pred.strip())
-            labels_text.append(label.strip())
-            input_text.append(input.strip())
-
-        metric = self.val_metric[dataloader_idx] if mode == 'validation' else self.test_metric[dataloader_idx]
-        assert len(preds_text) == len(labels_text) == len(input_text)
-        for _, (pred, label) in enumerate(zip(preds_text, labels_text)):
-            # To compute metrics like pearson or spearman correlation, we need to cast the predicted string and labels to floats.
-            pred, label = self.cast_for_metric(
-                pred=pred.strip(),
-                label=label.strip(),
-                metric_name=self.val_metric_name if mode == 'validation' else self.test_metric_name,
-                class_labels=self.cfg.data.validation_ds.metric.get('class_labels', None)
-                if mode == 'validation'
-                else self.cfg.data.test_ds.metric.get('class_labels', None),
-                labels_are_strings=self.cfg.data.validation_ds.metric.get('labels_are_strings', False)
-                if mode == 'validation'
-                else self.cfg.data.test_ds.metric.get('labels_are_strings', False),
-            )
-            _ = metric(pred, label)
-
-        return {
-            'loss': loss,
-            'preds': preds_text,
-            'labels': labels_text,
-            'inputs': input_text,
-        }
+        # if mode == 'validation':
```

Comment:
these lines are commented out since there's no `self.validation_step_outputs` in the model, maybe PTL2.0 has it